Source/Libraries/openXDA.Nodes/Types/Email/ScheduledEmailNode.cs

```diff
@@ -0,0 +1,214 @@
+
+            public ScheduledEmailWebController(ScheduledEmailNode node) =>
+                Node = node;
+
+            [HttpGet]
+            public HttpResponseMessage Schedules()
+            {
+                string status = Node.ScheduleManager.Status;
+                HttpResponseMessage response = new HttpResponseMessage();
+                response.Content = new StringContent(status);
+                return response;
+            }
+        }
+
+        #endregion
+
+        #region [ Constructors ]
+
+        public ScheduledEmailNode(Host host, Node definition, NodeType type)
+            : base(host, definition, type)
+        {
+            ScheduleManager = new ScheduleManager();
+            ScheduleManager.ScheduleDue += ScheduleManager_ScheduleDue;
+            ScheduleManager.Start();
+            Action<object> configurator = GetConfigurator();
+            OnReconfigure(configurator);
+        }
+
+        #endregion
+
+        #region [ Properties ]
+
+        private ScheduleManager ScheduleManager { get; }
+        private bool IsDisposed { get; set; }
+
+        #endregion
+
+        #region [ Methods ]
+
+        public override IHttpController CreateWebController() =>
+            new ScheduledEmailWebController(this);
+
+        public void Dispose()
+        {
+            if (IsDisposed)
+                return;
+
+            ScheduleManager.Dispose();
+            IsDisposed = true;
+        }
+
+        protected override void OnReconfigure(Action<object> configurator)
+        {
+            List<Schedule> schedules = ScheduleManager.Schedules;
+            lock (schedules) { schedules.Clear(); }
+
+            using (AdoDataConnection connection = CreateDbConnection())
+            using (DataTable table = connection.RetrieveData("SELECT ID, Name, Schedule FROM ScheduledEmailType"))
+            {
+                foreach (DataRow row in table.Rows)
+                {
+                    int id = row.ConvertField<int>("ID");
+                    string name = row.ConvertField<string>("Name");
+                    string scheduleRule = row.ConvertField<string>("Schedule");
+                    string scheduleName = $"[{id}] {name}";
+                    ScheduleManager.AddSchedule(scheduleName, scheduleRule);
+                }
+            }
+        }
+
+        private void ScheduleManager_ScheduleDue(object sender, GSF.EventArgs<Schedule> e)
+        {
+            const string Pattern = @"\[(\d+)\] (.*)";
+            string scheduleName = e.Argument.Name;
+            Match match = Regex.Match(scheduleName, Pattern);
+
+            if (!match.Success)
+                return;
+
+            string idMatch = match.Groups["1"].Value; //0th group is the entire string, 2nd is the name
+
+            if (!int.TryParse(idMatch, out int id))
+                return;
+
+            DateTime now = CurrentMinute();
+            DateTime currentScheduled = e.Argument.LastDueAt;
```

Comment:
This is good use of the `LastDueAt` property. We should get rid of `now` and the `CurrentMinute()` function, passing `currentScheduled` into the scheduled email process instead.